chroma_feedback/consumer/razer_chroma/device.py

```diff
@@ -68,5 +76,14 @@ def pulsate_device(device : Any, state : Dict[str, Any]) -> bool:
 	if device.has('brightness'):
 		device.brightness = state['brightness'][0]
 	if device.fx.has('logo') and device.fx.has('scroll'):
-		return device.fx.misc.logo.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2]) and device.fx.misc.scroll_wheel.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2])
+		if device.fx.has('logo_pulsate') and device.fx.has('scroll_pulsate'):
```

Comment:
One might ask why the check for logo/scroll pulsate is still here if the functions have been broken out and are selected by the device support block above.  The main reason is to take care for 'odd' devices which might not support one or the other.

Each function (breath, pulsate) now tries to perform only that effect, rather than falling back to the other one.